pkg/agent/core/ngt/handler/grpc/index_test.go

```diff
@@ -1029,106 +1029,874 @@ func Test_server_SaveIndex(t *testing.T) {
+				args: args{
+					c: &payload.Control_CreateIndexRequest{
+						PoolSize: uint32(insertCnt),
+					},
+				},
+				fields: fields{
+					srvOpts:   defaultSrvOpts,
+					svcCfg:    defaultSvcCfg,
+					svcOpts:   append(defaultSvcOpts, service.WithCopyOnWrite(true)),
+					indexPath: mkdirTemp(),
+				},
+				beforeFunc: func(t *testing.T, ctx context.Context, s Server, n service.NGT, test test) {
+					t.Helper()
+					var err error
+					if irs, err = request.GenMultiInsertReq(request.Float, vector.Gaussian, insertCnt, dim, defaultInsertConfig); err != nil {
+						t.Error(err)
+					}
+					if _, err := s.MultiInsert(ctx, irs); err != nil {
+						t.Error(err)
+					}
+				},
+				want: want{
+					wantRes: emptyPayload,
+				},
+				checkFunc: func(test test, ctx context.Context, s Server, n service.NGT, w want, e *payload.Empty, err error) error {
+					if err := defaultCheckFunc(test, ctx, s, n, w, e, err); err != nil {
+						return err
+					}
+					return checkBackupFolder(test.fields, ctx, irs.Requests)
+				},
+			}
+		}(),
+		func() test {
+			insertCnt := 100
+			var irs *payload.Insert_MultiRequest
+
+			return test{
+				name: "Decision Table Testing case 3.4: success to create and save index with poolSize = 0",
+				args: args{
+					c: &payload.Control_CreateIndexRequest{
+						PoolSize: 0,
+					},
+				},
+				fields: fields{
+					srvOpts:   defaultSrvOpts,
+					svcCfg:    defaultSvcCfg,
+					svcOpts:   append(defaultSvcOpts, service.WithCopyOnWrite(true)),
+					indexPath: mkdirTemp(),
+				},
+				beforeFunc: func(t *testing.T, ctx context.Context, s Server, n service.NGT, test test) {
+					t.Helper()
+					var err error
+					if irs, err = request.GenMultiInsertReq(request.Float, vector.Gaussian, insertCnt, dim, defaultInsertConfig); err != nil {
+						t.Error(err)
+					}
+					if _, err := s.MultiInsert(ctx, irs); err != nil {
+						t.Error(err)
+					}
+				},
+				want: want{
+					wantRes: emptyPayload,
+				},
+				checkFunc: func(test test, ctx context.Context, s Server, n service.NGT, w want, e *payload.Empty, err error) error {
+					if err := defaultCheckFunc(test, ctx, s, n, w, e, err); err != nil {
+						return err
+					}
+					return checkBackupFolder(test.fields, ctx, irs.Requests)
+				},
+			}
+		}(),
+	}
+
+	for _, tc := range tests {
+		test := tc
+		t.Run(test.name, func(tt *testing.T) {
+			tt.Parallel()
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			afterFunc := test.afterFunc
+			if test.afterFunc == nil {
+				afterFunc = defaultAfterFunc
+			}
+			defer afterFunc(tt, test)
+
+			checkFunc := test.checkFunc
+			if test.checkFunc == nil {
+				checkFunc = defaultCheckFunc
+			}
+
+			eg, _ := errgroup.New(ctx)
+			ngt, err := service.New(test.fields.svcCfg, append(test.fields.svcOpts,
+				service.WithErrGroup(eg),
+				service.WithIndexPath(test.fields.indexPath))...)
```

Comment:
```suggestion
			ngt, err := service.New(
				test.fields.svcCfg,
				append(
					test.fields.svcOpts,
					service.WithErrGroup(eg),
					service.WithIndexPath(test.fields.indexPath),
				)
			...)
```
Could you format for more readable ?🤔

Reply:
fixed